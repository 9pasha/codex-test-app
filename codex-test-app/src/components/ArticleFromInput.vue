<template>
  <v-container>
    <p>Canvas size: <span class="indigo--text">{{ canvasSize }}</span></p>
    <p>Line: <span class="indigo--text">{{ line }}</span></p>
    <p>Rectangle: <span class="indigo--text">{{ rectangle }}</span></p>
    <p>Bucket fill: <span class="indigo--text">{{ fillBucket }}</span></p>
  </v-container>
</template>

<script>
import Input from "!raw-loader!../../../input.txt";
export default {
  name: "ArticleFromInput",
  data: () => ({
    text: Input,
  }),
  computed: {
    canvasSize() {
      const reg = /C \d+ \d+/gi;
      return this.text.match(reg).join("; ");
    },
    line() {
      const reg = /L \d+ \d+ \d+ \d+/gi;
      return this.text.match(reg).join("; ");
    },
    rectangle() {
      const reg = /R \d+ \d+ \d+ \d+/gi;
      return this.text.match(reg).join("; ");
    },
    fillBucket() {
      const reg = /B \d+ \d+ \w/gi;
      return this.text.match(reg).join("; ");
    }
  }
}
</script>

<style scoped>

</style>
