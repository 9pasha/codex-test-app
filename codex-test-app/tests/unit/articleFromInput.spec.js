import { shallowMount, createLocalVue } from "@vue/test-utils";
import ArticleFromInput from "@/components/ArticleFromInput.vue";
import Vue from "vue";
import Vuex from "vuex";
import Vuetify from "vuetify";

const localVue = createLocalVue();

localVue.use(Vuex);
Vue.use(Vuetify);

describe("ArticleFromInput.vue", () => {
  let vuetify;
  let store;
  let actions;

  beforeEach(() => {
    vuetify = new Vuetify();
    actions = {
      draw: jest.fn()
    };
    store = new Vuex.Store({
      state: {
        canvas: {
          x: 5,
          y: 10
        },
        isCorrectCanvas: false
      },
      mutations: {
        updateLine: jest.fn(),
        updateRect: jest.fn(),
        updateFillBucket: jest.fn(),
        updateCanvas: jest.fn(),
        setOperations: jest.fn(),
        updateIsCorrectCanvas: jest.fn()
      },
      actions
    });
  });
  it("emulate button click", () => {
    const wrapper = shallowMount(ArticleFromInput, {
      localVue,
      vuetify,
      store
    });
    const button = wrapper.find(".save-button");

    button.trigger("click");
    expect(actions.draw).toHaveBeenCalled();
  });
  it("should correct view data", async () => {
    const wrapper = shallowMount(ArticleFromInput, {
      localVue,
      vuetify,
      store
    });

    await wrapper.setData({
      text: "C 20 4\nL 1 2 6 2\nL 6 3 6 4\nR 16 1 20 3\nB 10 2 o"
    });
    expect(wrapper.vm.text).toBe(
      "C 20 4\nL 1 2 6 2\nL 6 3 6 4\nR 16 1 20 3\nB 10 2 o"
    );

    const articles = wrapper.findAll("span");

    expect(articles).toHaveLength(4);
    expect(articles.at(0).text()).toEqual("C 20 4");
    expect(articles.at(1).text()).toEqual("L 1 2 6 2; L 6 3 6 4");
    expect(articles.at(2).text()).toEqual("R 16 1 20 3");
    expect(articles.at(3).text()).toEqual("B 10 2 o");
  });
  it("should match snapshots", () => {
    const wrapper = shallowMount(ArticleFromInput, {
      localVue,
      vuetify,
      store
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
