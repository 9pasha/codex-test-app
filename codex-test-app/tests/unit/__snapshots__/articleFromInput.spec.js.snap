// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleFromInput.vue should match snapshots 1`] = `
<v-container-stub tag="div">
  <p>
    Canvas size: <span class="indigo--text">C 20 4</span></p>
  <p>
    Line: <span class="indigo--text">L 1 2 6 2; L 6 3 6 4</span></p>
  <p>
    Rectangle: <span class="indigo--text">R 16 1 20 3</span></p>
  <p>
    Bucket fill: <span class="indigo--text">B 10 2 o</span></p>
  <v-btn-stub color="primary mt-5" elevation="2" tag="button" activeclass="" large="true" type="button" class="save-button">
    Save to .txt
  </v-btn-stub>
</v-container-stub>
`;
